### PR TITLE
feat: add savings to price comps

### PR DESCRIPTION
**Pull request by xianny**
# Description

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/utils/price_comparison_utils.ts [hunk 1]
```diff
@@ -53,139 +54,148 @@ const emptyPlaceholderSources = Object.values(ERC20BridgeSource).reduce<SourceCo
...
+    buyAmount: BigNumber;
+    sellAmount: BigNumber;
+    sellTokenToEthRate: BigNumber;
+    buyTokenToEthRate: BigNumber;
+    gasPrice: BigNumber;
+    estimatedGas: BigNumber;
+    quoteReport?: QuoteReport;
+}
+
+function getPriceComparisonFromQuoteOrThrow(
```
Comment (kimpers) [inline src/utils/price_comparison_utils.ts:90]: NIT: This fn never intentionally throws so the naming here might be a bit misleading. I originally put it inside a try/catch to guard against unexpected run-time errors in the calculations erroring out the entire quote request. It should never (🙏 ) error out under normal circumstances.

**Review by kimpers — APPROVED** [review]
Looks great! Awesome work on this! It has been a pretty dreadful process but I think we have a really solid solution now 🙏

**Comment by xianny** [comment]
deploy staging

**Comment by xianny** [comment]
Screen Shot 2020-12-07 at 1 35 21 PM
Screen Shot 2020-12-07 at 1 35 41 PM
Screen Shot 2020-12-07 at 1 37 18 PM

**Comment by xianny** [comment]
deploy staging

**Comment by xianny** [comment]
Screen Shot 2020-12-08 at 4 12 42 PM
Screen Shot 2020-12-08 at 4 12 59 PM
Screen Shot 2020-12-08 at 4 13 30 PM
 Looking fine to me

**Pull request merged**